devito/ir/clusters/cluster.py

```diff
@@ -409,7 +414,8 @@ def dtype(self):
         elif len(dtypes) == 1:
             return dtypes.pop()
         else:
-            raise ValueError("Unsupported ClusterGroup [mixed integer arithmetic ?]")
+            # E.g., mixed integer arithmetic
+            return max(dtypes, key=lambda i: np.dtype(i).itemsize)
```

Comment:
np.find_common_type(fdtypes, []) for the whole id/else

Reply:
This looked cool and promising... but then I tried it, and I'm not sure this works for us anymore. Consider the following:

```
In [6]: np.find_common_type([np.int32, np.float32], [np.float32])
Out[6]: dtype('float64')

In [7]: np.find_common_type([], [np.int32, np.float32])
Out[7]: dtype('float64')

In [8]: np.find_common_type([np.int32, np.float32], [])
Out[8]: dtype('float64')
```

We clearly want `np.float32` here, not `np.float64`.

Reply:
Interesting, numpy seem to say that an int32 cannon e converted to a float32 safely so that's why it returns float64
```
In [14]: np.can_cast(np.int32, np.float32)
Out[14]: False
```

So I'm not sure of it's a numpy internal thing but yeah definitely not what we want.